torch/nn/modules/linear.py

```diff
@@ -178,4 +179,62 @@ def extra_repr(self) -> str:
             self.in1_features, self.in2_features, self.out_features, self.bias is not None
         )
 
+
+class LazyLinear(LazyModuleMixin, Linear):
+    r"""A :class:`torch.nn.Linear` module with lazy initialization.
+
+    In this module, the `weight` and `bias` are of :class:`torch.nn.Uninitialized`
+    class. They will be initialized  after the first call to ``forward`` is done.
+    The parameters are materialized in the `initialize_parameters` method, which
+    will instantiate them with any device or datatype the module is set to.
+    After the dry run, the module will become a :class:`torch.nn.Linear`
```

Comment:
This last sentence is work keeping, since it's not duplicative with the lazy module documentation, but it needs a small tweak:

"After the first call to forward this module will instantiate its uninitialized parameters and become a regular :class:`torch.nn.Linear` module."